generators/tracegen/src/main/scala/Tile.scala

```diff
@@ -29,6 +34,180 @@ class TraceGenTile(val id: Int, val params: TraceGenParams, q: Parameters)
   override lazy val module = new TraceGenTileModuleImp(this)
 }
 
+class BoomTraceGenTile(val id: Int, val params: TraceGenParams, q: Parameters)
+  extends BaseTile(params, SynchronousCrossing(), HartsWontDeduplicate(params), q) {
+  val boom_params = p.alterMap(Map(TileKey -> BoomTileParams(
+    dcache=params.dcache,
+    core=BoomCoreParams(nPMPs=0, numLdqEntries=32, numStqEntries=32, useVM=false))))
+  val dcache = params.dcache.map {
+    dc => LazyModule(new BoomNonBlockingDCache(hartId)(boom_params))
+  }.get
+
+  val intInwardNode: IntInwardNode = IntIdentityNode()
+  val intOutwardNode: IntOutwardNode = IntIdentityNode()
+  val slaveNode: TLInwardNode = TLIdentityNode()
+  val ceaseNode: IntOutwardNode = IntIdentityNode()
+  val haltNode: IntOutwardNode = IntIdentityNode()
+  val wfiNode: IntOutwardNode = IntIdentityNode()
+
+  val masterNode = visibilityNode
+  masterNode := dcache.node
+
+  override lazy val module = new BoomTraceGenTileModuleImp(this)
+}
+
+class BoomTraceGenTileModuleImp(outer: BoomTraceGenTile)
+  extends BaseTileModuleImp(outer) with MemoryOpConstants {
+
+  val status = IO(new GroundTestStatus)
+
+  val tracegen = Module(new TraceGenerator(outer.params))
+  tracegen.io.mem := DontCare
+  tracegen.io.hartid := constants.hartid
+
+  val ptw = Module(new DummyPTW(1))
+
+  val lsu = Module(new LSU()(outer.boom_params, outer.dcache.module.edge))
+  lsu.io.core.tsc_reg := 0.U(1.W)
+  ptw.io.requestors.head <> lsu.io.ptw
+  outer.dcache.module.io.lsu <> lsu.io.dmem
+
+
+  val rob_sz = outer.boom_params(TileKey).core.asInstanceOf[BoomCoreParams].numRobEntries
```

Comment:
I think it would be better to put this glue logic into a separate module that just gets instantiated in the TileModuleImp.